msprime/demography.py

```diff
@@ -643,11 +644,19 @@ def from_old_style(pop_config, Ne=1):
         initial_size = (
             Ne if pop_config.initial_size is None else pop_config.initial_size
         )
-        return Population(
+        population = Population(
             initial_size=initial_size,
             growth_rate=pop_config.growth_rate,
-            extra_metadata=pop_config.metadata,
         )
+        metadata = pop_config.metadata
+        if metadata is not None and isinstance(metadata, dict):
```

Comment:
The ``isinstance(metadata, dict)`` is a bit too strict as this will rule out other valid mapping types (like defaultdict, probably). Either change to ``collections.abc.Mapping`` or drop the check - it's probably better to just fail here if someone adds a non-dict argument?

Reply:
The extra metadata can be a byte array, I think? At least, that's what I saw before having the check. I'll change it to a `abc.Mapping`.